scrapy/extensions/robotstxtparser.py

```diff
@@ -0,0 +1,130 @@
+import sys
+import logging
+from six.moves.urllib_robotparser import RobotFileParser
+
+from scrapy.utils.python import to_native_str, to_unicode
+
+logger = logging.getLogger(__name__)
+
+class PythonRobotParser():
+    def __init__(self, content, crawler):
```

Comment:
I think we can make it a Scrapy component like others, with from_crawler method, and pass just `spider` to `__init__` method (it is needed only for logging, right?).

Reply:
Yes, it is used only for logging. I have made the change.